src/drift.core/model/TeamProject.cs

```diff
@@ -15,5 +16,17 @@ public class TeamProject
         public IList<Team> Teams { get; set; } = new List<Team>();
 
         public TeamProjectStatus Status { get; set; } = TeamProjectStatus.Active;
+
+        public string GenerateDriftReport()
+        {
+            var result = string.Empty;
+
+            foreach (var pattern in this.Patterns)
+            {
+                result += pattern.GenerateDriftReport(this);
+            }
```

Comment:
How about a yield return without creating temp collections?

Reply:
Rewritten to return an IEnumerable<Deviation>. We need for .net core 3.0 for async streams, not available yet.